lib/solidus_shipstation/api/syncability_checker.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module SolidusShipstation
+  module Api
+    class SyncabilityChecker
+      class << self
+        def call(shipment)
+          return false unless shipment.order.completed?
+
+          !!(shipment_requires_creation?(shipment) || shipment_requires_update?(shipment))
+        end
+
+        private
+
+        def shipment_requires_creation?(shipment)
+          shipment.shipstation_synced_at.nil? &&
+            Time.zone.now - shipment.created_at < SolidusShipstation.config.api_sync_threshold
```

Comment:
This looks fragile if I got it correctly, how do we know that the shipment is not in a jobs queue even if those checks are both true?